pkg/services/container/morph/executor.go

```diff
@@ -274,3 +291,41 @@ func (s *morphExecutor) GetExtendedACL(_ context.Context, body *container.GetExt
 
 	return res, nil
 }
+
+func (s *morphExecutor) validateToken(t *sessionV2.Token, cID *refs.ContainerID, op sessionV2.ContainerSessionVerb) error {
+	c := t.GetBody().GetContext()
+	cc, ok := c.(*sessionV2.ContainerSessionContext)
+	if !ok {
+		return errors.New("session is not container related")
+	}
+
+	if verb := cc.Verb(); verb != op {
+		return fmt.Errorf("wrong container session operation: %s", verb)
+	}
+
+	if cID == nil { // can be nil for PUT
+		return nil
+	}
+
+	if sessionCID := cc.ContainerID().GetValue(); !bytes.Equal(sessionCID, cID.GetValue()) {
+		return fmt.Errorf("wrong container: %s", base58.Encode(sessionCID))
+	}
+
+	var cIDSDK cid.ID
+	err := cIDSDK.ReadFromV2(*cID)
+
+	if err != nil {
+		return fmt.Errorf("cID converting: %w", err)
+	}
+
+	cnr, err := s.rdr.Get(cIDSDK)
+	if err != nil {
+		return fmt.Errorf("reading container from the network: %w", err)
+	}
+
+	if issuer := t.GetBody().GetOwnerID().GetValue(); !bytes.Equal(cnr.Value.Owner().WalletBytes(), issuer) {
+		return fmt.Errorf("session session was issued not by the owner, issuer: %x", issuer)
+	}
+
+	return nil
```

Comment:
why are not all fields checked?

Reply:
what else fields to check? that is not IR validation before we try to change smth, just a logical validation that you are not trying to make totally incorrect things (container owner do not know about your operation but you still trying it). did not want to copy IR behavior

epoch can not be ensured: the final check will be done on the IR side, no one knows at what state and when

Reply:
there are other stateless conditions like `lifetime` field presence, `signature` correctness, etc. And some stateful conditions are consistent, e.g. expiraiton claim

Reply:
> there are other stateless conditions like lifetime field presence

i would say it is more like a protocol validation (if we are talking about the presence, and epoch values i wouldn't check ever by the intermediate SN ever, as I've already said)

> signature correctness

well... ok, done but not valuable to me

Reply:
> well... ok, done but not valuable to me

words of some undefined behavior...